lang/LangPrimSource/PyrPrimitive.cpp

```diff
@@ -108,13 +108,14 @@ int slotStrLen(PyrSlot* slot) {
     return -1;
 }
 
-int slotStrVal(PyrSlot* slot, char* str, int maxlen) {
+int slotStrVal(PyrSlot* slot, char* str, int maxSize) {
```

Comment:
Should we be checking for negative values here as passing a negative to std::memcpy will result in converting it to `size_t`, resulting in a large amount of memory being copied? Better yet, this could be a size_t in the signature?

I know this is incorrect usage of the api, and supercollider doesn't do these checks anywhere else, but making the signature a size_t seems like an easy win because you also get the compiler warnings which can be fixed later.

Reply:
> Better yet, this could be a size_t in the signature?

Yeah, this should really be a `size_t`. Though this wouldn't make the code any safer because if the upstream value is negative, the modulo will just happen here instead of instead `memcpy`. I think we need to trust the caller to use the function correctly. Range checks should only be necessary for user input.

That being said, I can change this to `size_t`, just to set a good example :)

Reply:
I think that is a good idea, because it means you get implicit conversion warnings higher up where you can actually do something about them, or at least reminding users of this function that the conversion will take place.

Another option would be something like `std::span<char>` providing debug range checks ... but that is too much of a change for this...

Reply:
Changed to `size_t`!